tests/estimators/object_detection/test_object_seeker.py

```diff
@@ -0,0 +1,186 @@
+    from pytorchyolo.utils.loss import compute_loss
+
+    from art.estimators.object_detection.pytorch_yolo import PyTorchYolo
+
+    model_path = "/tmp/PyTorch-YOLOv3/config/yolov3.cfg"
+    weights_path = "/tmp/PyTorch-YOLOv3/weights/yolov3.weights"
+    model = models.load_model(model_path=model_path, weights_path=weights_path)
+
+    class YoloV3(torch.nn.Module):
+        def __init__(self, model):
+            super().__init__()
+            self.model = model
+
+        def forward(self, x, targets=None):
+            if self.training:
+                outputs = self.model(x)
+                loss, _ = compute_loss(outputs, targets, self.model)
+                loss_components = {"loss_total": loss}
+                return loss_components
+            else:
+                return self.model(x)
+
+    model = YoloV3(model)
+
+    params = [p for p in model.parameters() if p.requires_grad]
+    optimizer = torch.optim.SGD(params, lr=0.01)
+
+    object_detector = PyTorchYolo(
+        model=model,
+        input_shape=(3, 416, 416),
+        optimizer=optimizer,
+        clip_values=(0, 1),
+        channels_first=True,
+        attack_losses=("loss_total",),
+    )
+
+    (_, _), (x_test_cifar10, _) = get_default_cifar10_subset
+
+    x_test = cv2.resize(
+        x_test_cifar10[0].transpose((1, 2, 0)), dsize=(416, 416), interpolation=cv2.INTER_CUBIC
+    ).transpose((2, 0, 1))
+    x_test = np.expand_dims(x_test, axis=0)
+    x_test = np.repeat(x_test, repeats=2, axis=0)
+
+    # Create labels
+
+    result = object_detector.predict(x=x_test)
+
+    y_test = [
+        {
+            "boxes": result[0]["boxes"],
+            "labels": result[0]["labels"],
+            "scores": np.ones_like(result[0]["labels"]),
+        },
+        {
+            "boxes": result[1]["boxes"],
+            "labels": result[1]["labels"],
+            "scores": np.ones_like(result[1]["labels"]),
+        },
+    ]
+
+    yield object_detector, x_test, y_test
+
+
+@pytest.mark.only_with_platform("pytorch")
+def test_pytorch_train(art_warning, get_pytorch_yolo):
+    object_detector, x_test, y_test = get_pytorch_yolo
+    object_seeker = PyTorchObjectSeeker(
+        model=object_detector.model,
+        input_shape=object_detector.input_shape,
+        channels_first=object_detector.channels_first,
+        optimizer=object_detector.optimizer,
+        clip_values=object_detector.clip_values,
+        attack_losses=object_detector.attack_losses,
+        num_lines=3,
+        confidence_threshold=0.3,
+        iou_threshold=0.4,
+        prune_threshold=0.5,
+        device_type="cpu",
+    )
+
+    try:
+        # Compute loss before training
+        loss1 = object_seeker.compute_loss(x=x_test, y=y_test)
+
+        # Train for one epoch
+        object_seeker.fit(x_test, y_test, nb_epochs=1)
+
+        # Compute loss after training
+        loss2 = object_seeker.compute_loss(x=x_test, y=y_test)
+
+        assert loss1 != loss2
```

Comment:
Should `loss2` be smaller than `loss1`?

Reply:
I borrowed the same unit test from `test_pytorch_yolo.py`. Logically it should be less than, but I believe it was not equal in case of stochasticity that causes the loss to increase. I can look into this and see if it can be changed to less than for both tests.